### ISSUE TITLE
Formatting .liquid files deletes their contents

### ISSUE THREAD

**Issue by gregbarbosa** [Shopify/theme-tools#327]
Type: <b>Bug</b>

Selecting the Format Document in the Command Palette causes all code in the open editor to be deleted, instead of formatting.

Extension version: 2.1.3
VS Code version: Code 1.87.2 (Universal) (863d2581ecda6849923a2118d93a088b0745d9d6, 2024-03-08T15:21:31.043Z)
OS version: Darwin arm64 23.4.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|5, 6, 6|
|Memory (System)|16.00GB (0.65GB free)|
|Process Argv|--disable-extensions --crash-reporter-id 4fe9d41b-ab3b-4fee-a441-35559309088a --crash-reporter-id 4fe9d41b-ab3b-4fee-a441-35559309088a|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes516:30244333
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
vscaat:30438848
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
pydisww2:30991134
cppperfcontrol:30979541
d34g3935:30971562
fegfb526:30981948
bg6jg535:30979843
ccp2r3:30993541
dsvsc020:30976470
pythonaic:30992952

```

</details>

**Comment by charlespwd** [comment]
Would need more details since it appears to be running fine for most of us. Can you provide the following:

- List of Liquid related VS Code extensions
- `settings.json` contents

Here are my settings for Liquid files:

```jsonc
{
  "[liquid]": {
    "editor.defaultFormatter": "Shopify.theme-check-vscode",
    "editor.formatOnSave": true
  },
}
```

https://github.com/Shopify/theme-tools/assets/4990691/e998124d-455b-4d00-9726-bd72bfb1f2fa

Feel free to reopen if you still have the issue after debugging your setup.

**Comment by charlespwd** [comment]
That is _so_ strange... Wondering if that's a problem with the input file? Does this happen on any file or only that one?

**Comment by gregbarbosa** [comment]
Fixed with a fresh yarn install through Homebrew. Thanks @charlespwd.

Here's what I did:
1. Narrowed the problem down to my local environment by testing with a new local user and GitHub Codespaces.
2. Removed all Homebrew packages and dependencies.
3. Completely uninstalled VS Code and Homebrew.
4. Cleaned up and removed all PATH variables in `.zshrc`, `.bashrc`, and `.bash_profile`
5. Reinstalled VS Code and Homerbiew.
6. Reinstalled each Homebrew package one by one, checking the formatter's behavior each time.
7. The problem was fixed immediately after a fresh yarn install.

Looks like the issue was a yarn misconfiguration interfering with prettier, possibly because I hadn't installed yarn through Homebrew before.

**Comment by Afamsinachi** [comment]
Here is what worked for me on Windows
1. I uninstalled ruby and all the shopify development kit, including the Shopify Liquid vscode extension.
2. I cleared vscode cache in AppData\Roaming\code\cachedData directory
3. Reinstalled Ruby and the shopify development kit and Shopig Liquid vscode extension 
4. The problem was fixed after all the process

**Comment by gregbarbosa** [comment]
@joshistoast Did my solution above help you at all? My problem was with my `yarn` install.

**Comment by joshistoast** [comment]
> @joshistoast Did my solution above help you at all? My problem was with my `yarn` install.

Just uninstalled and reinstalled `yarn` via brew and cleared it's cache files, no difference in formatting behavior.

**Comment by gregbarbosa** [comment]
That’s frustrating. My issue was because I initially installed `yarn` using npm (checked using `which yarn`). When I switched to installing it via Homebrew, the problem was fixed.

It seems like the formatter might be searching for dependencies in specific directories and if it doesn’t find them, it defaults to wiping everything out. I stumbled on this while troubleshooting using `formatOnSave` you mentioned in the other issue.
